File: backend/src/main/java/com/example/backend/entity/Order.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String status = "Pending";

    private Double totalAmount;

    private String shippingAddressLine;
    private String shippingCity;
    private String shippingState;
    private String shippingPostalCode;
    private String shippingCountry;

    private LocalDateTime orderDate = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "payment_id")
    private Payment payment;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> orderItems;

    @OneToOne(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private ShipmentTracking shipmentTracking;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Complaint> complaints;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: frontend/src/app/shared/models/order.ts
import { User } from './user';
import { OrderItem } from './order-item';
import { OrderStatus } from './order-status';

export interface Order {
  id: number;
  userId: number;
  paymentId?: number;
  totalAmount: number;
  status: OrderStatus;
  shippingAddressLine: string;
  shippingCity: string;
  shippingState: string;
  shippingPostalCode: string;
  shippingCountry: string;
  orderDate: Date;
  orderItems?: OrderItem[];
  user?: User;
}



